docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,357 @@
+  because it is not supported.
+
+Commit `f66ac0be1c19 <https://github.com/scality/metalk8s/commit/f66ac0be1c191be8fa31a925c28d34c113eb172c>`_
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+.. code:: text
+
+    build: fix concurrent build on MacOS
+
+    When trying to use the parallel execution feature of `doit` on Mac, we
+    observe that the worker processes are killed by the OS and only the
+    main one survives.
+
+    The issues seems related to the fact that:
+    - by default `doit` uses `fork` (through `multiprocessing`) to spawn its
+      workers
+    - since macOS 10.13 (High Sierra), Apple added a new security measure[1]
+      that kill processes that are using a dangerous mix of threads and
+      forks[2])
+
+    As a consequence, now instead of working most of the time (and failing
+    in a hard way to debug), the processes are directly killed.
+
+    There are three ways to solve this problems:
+    1. set the environment variable `OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES.`
+    2. don't use `fork`
+    3. fix the code that uses a dangerous mix of thread and forks
+
+    ① is not good as it doesn't fix the underlying issue: it only disable
+    the security and we're back to "works most of the time, sometimes does
+    weird things"
+    ② is easy to do because we can tell to `doit` to uses only threads
+    instead of forks.
+    ③ is probably the best, but requires more troubleshooting/time/…
+
+    In conclusion, this commit implements ② until ③ is done (if ever…) by
+    detecting macOS and forcing the use of threads in that case.
+
+    [1]: http://sealiesoftware.com/blog/archive/2017/6/5/Objective-C_and_fork_in_macOS_1013.html
+    [2]: https://blog.phusion.nl/2017/10/13/why-ruby-app-servers-break-on-macos-high-sierra-and-what-can-be-done-about-it/
+
+    Closes: #1354
+
+Some things to note about this commit message:
+
+- Observed problem is described: parallel builds crash on macOS.
+
+- Root cause is analyzed: OS security measure + thread/fork mix.
+
+- Several solution are proposed: disable the security, workaround the problem
+  or fix the root cause.
+
+- Selection of a solution is explained: we go for the workaround because it is
+  easy and faster.
+
+- Extra-references are given: links in the footer gives more in-depth
+  explanations/context.
+
+Conclusion
+~~~~~~~~~~
+
+When reviewing a change, do not simply look at the correctness of the code:
+review the commit message itself and request improvements to its content.
+Look out for commits that can be divided, ensure that cosmetic changes are not
+mixed with functional changes, …
+
+The goal here is to improve the long term maintainability, by a wide variety of
+developers. Considering that we are an Open Source project, external
+contributors may only have the Git history to get some context so it is
+important to have a useful Git history.
```

Comment:
This doc should be used by open source contributors as well. IMO we should not talk about the fact that we are an "Open Source project" in the doc. 
I would suggest something 
```suggestion
The goal here is to improve the long term maintainability, by a wide variety of
developers who may only have the Git history to get some context so it is
important to have a useful Git history.
```

(Suggestion does not work very well with multi line comment :stuck_out_tongue: )

Reply:
Fair enough. Done!